index.bs

```diff
@@ -10600,129 +10606,162 @@ can restart executing this algoritm if needed.
-			<a href="#channel-up-mixing-and-down-mixing">Up or down-mix</a>
-			it to match if number of input channels of this
-			{{AudioNode}}.
+				3. <a>Queue a control message</a> to set the {{[[current value]]}} slot
+					of this {{AudioParam}} to the last value computed in the preceding
+					step.
 
-		3. If this {{AudioNode}} is a <a>source node</a>, [=Computing a block of audio|compute a block of audio=],
-			and <a href="#available">make it available for reading</a>.
+			2. If this {{AudioNode}} has any {{AudioNode}}s connected to its input,
+				<a href="#SummingJunction">sum</a> the buffers
+				<a href="#available-for-reading">made available for reading</a> by all
+				{{AudioNode}}s connected to this {{AudioNode}}. The resulting buffer is
+				called the <dfn>input buffer</dfn>.
+				<a href="#channel-up-mixing-and-down-mixing">Up or down-mix</a> it to
+				match if number of input channels of this {{AudioNode}}.
 
-		4. Else, if this {{AudioNode}} is a <a>destination node</a>,
-			[=Recording the input|record the input=] of this {{AudioNode}}.
+			3. If this {{AudioNode}} is a <a>source node</a>,
+				[=Computing a block of audio|compute a block of audio=], and
+				<a href="#available-for-reading">make it available for reading</a>.
 
-		5. Else, <a>process</a> the <a>input buffer</a>,
-			and <a href="#available">make available for reading</a> the resulting buffer.
+			4. If this {{AudioNode}} is an {{AudioWorkletNode}}, invoke associated
+				{{AudioWorkletProcessor}}'s <a href="#audioworkletprocessor-process">
+				process</a> method to <a href="computing-block">compute a block of
+				audio</a> with the argument of <a href="#">input buffer</a>, output
+				buffer and <a href="#input-audioparam-buffer">input AudioParam buffer
+				</a>.
 
-	9. <a>Atomically</a> perform the following steps:
-		1. Increment {{[[current frame]]}} by the [=render quantum size=].
-		1. Set {{BaseAudioContext/currentTime}} to {{[[current frame]]}} divided by {{BaseAudioContext/sampleRate}}.
+				1. Any {{Promise}} resolved within the execution of process method will
+					be queued into the microtask queue in the {{AudioWorkletGlobalScope}}.
 
-	10. Return true.
-</div>
+			5. Else, if this {{AudioNode}} is a <a>destination node</a>,
+				<a href="record-input">record the input</a> of this {{AudioNode}}.
+
+			6. Else, <a>process</a> the <a>input buffer</a>, and
+				<a href="#available-for-reading">make available for reading</a> the
+				resulting buffer.
+
+		6. <a>Atomically</a> perform the following steps:
+
+			1. Increment {{[[current frame]]}} by the [=render quantum size=].
+
+			2. Set {{BaseAudioContext/currentTime}} to {{[[current frame]]}} divided
+				by {{BaseAudioContext/sampleRate}}.
+
+		7. Set <var>RenderResult</var> to true.
+
+	4. <a href="https://html.spec.whatwg.org/multipage/webappapis.html#perform-a-microtask-checkpoint">
+		Perform a microtask checkpoint</a>.
+
+	5. Return <var>RenderResult</var>.
+	</div>
```

Comment:
This div closes the algorithm div right?  Then I think it should be in column 1, not indented.

Reply:
Done.